src/emmy/numerical/ode.cljc

```diff
@@ -249,11 +289,13 @@
   state derivative (and its argument package) from [0 to t1] in steps
   of size dt"
   [state-derivative state-derivative-args initial-state t1 dt]
-  (let [I (make-integrator state-derivative state-derivative-args)
-        out (atom [])
-        collector (fn [_ state]
-                    (swap! out conj state))]
-    (I initial-state dt t1 {:compile? true
-                            :epsilon 1e-6
-                            :observe collector})
-    @out))
+  (let [opts (integration-opts state-derivative
+                               state-derivative-args
+                               initial-state
+                               {})
+        f (:integrator opts)
+        v (mapv f (for [x (range 0 (+ t1 dt) dt)
```

Comment:
style wise you could do this in a `try, finally` , but treat that as me musing, not a request

Reply:
oh, I see what you mean. I'll fix it. That or a `with-stream-integrator` macro.